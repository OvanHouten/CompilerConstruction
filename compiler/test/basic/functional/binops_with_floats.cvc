extern void printInt(int integer);
extern void printFloat(float var);
extern void printNewlines(int num);

export int main() {
    float b = 2.3;
    float c;
    float d;
    int e;
    float f;
    printFloat(b);
    printNewlines(1);
    c = b + 4.2;
    printFloat(c);
    printNewlines(1);
    d = 5.2 * 6.2;
    printFloat(d);
    printNewlines(1);
    e = 17 % 8;
    printInt(e);
    printNewlines(1);
    f = c + d - (float) e * b;
    printFloat(f);
    printNewlines(2);
    return 0;
}
