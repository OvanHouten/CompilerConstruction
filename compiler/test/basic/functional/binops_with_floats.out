2.300000
6.500000
32.239998
1
36.439998

